client/my-sites/earn/memberships/index.jsx

```diff
@@ -301,6 +286,62 @@ class MembershipsSection extends Component {
 		);
 	}
 
+	getWording( subscriber ) {
+		const isOneTime = subscriber?.plan?.renew_interval === 'one-time';
+		const subscriber_email = subscriber?.user.user_email ?? '';
+		const wording = {
+			donation: {
+				button: isOneTime
+					? this.props.translate( 'Remove' )
+					: this.props.translate( 'Cancel Donation' ),
+				confirmation_subheading: this.props.translate( 'Do you want to remove this donation?' ),
+				confirmation_info:
+					this.props.translate(
+						'Removing this donation means that the user %(subscriber_email)s will no longer have access to any service granted by this plan, such as restricted premium content or digital subscription.',
+						{ args: { subscriber_email: subscriber_email } }
+					) +
+					' ' +
+					( isOneTime
+						? this.props.translate( 'The donation will not be refunded.' )
+						: this.props.translate(
+								'Donations already made will not be refunded to the user but any scheduled future donations will not be made.'
+						  ) ),
+				success: this.props.translate( 'Donation cancelled for %(subscriber_email)s.', {
+					args: { subscriber_email: subscriber_email },
+				} ),
+			},
+			purchase: {
+				button: isOneTime
+					? this.props.translate( 'Remove' )
+					: this.props.translate( 'Cancel Subscription' ),
+				confirmation_subheading: this.props.translate( 'Do you want to remove this purchase?' ),
+				confirmation_info:
+					this.props.translate(
+						'Removing this purchase means that the user %(subscriber_email)s will no longer have access to any service granted by this plan, such as restricted premium content or digital subscription.',
+						{ args: { subscriber_email: subscriber?.user.user_email ?? '' } }
+					) +
+					' ' +
+					( isOneTime
+						? this.props.translate( 'The purchase will not be refunded.' )
+						: this.props.translate(
+								'Payments already made will not be refunded to the user but any scheduled future payments will not be made.'
+						  ) ),
+				success: this.props.translate( 'Subscription cancelled for %(subscriber_email)s.', {
+					args: { subscriber_email: subscriber_email },
+				} ),
+			},
+		};
+
+		// plan.type is null for non-donations, it's not known what they are, so they're called purchase here.
+		// That is also used as the fallback in case other plan types are added in the future without updating
```

Comment:
Wouldn't it be better to fail if we have a subscriber.plan.type with no entry in the wording map? With the current approach, we will be loading some wording even though we haven't configured it, this can lead to human error that could be easily avoided during the development phase by failing early.

Reply:
The behaviour is consistent with every other part of the earn/payment-blocks output, it always falls back to the default wording (or rather it special cases the donation wording, but it's the same effect).

The wording is unlikely to be totally wrong, the impact of any human error would be negligible.